sqlalchemy_exasol/pyodbc.py

```diff
@@ -164,5 +167,139 @@ def is_disconnect(self, e, connection, cursor):
+            connection=connection,
+            schema=schema,
+            table_name=table_name,
+            **kw
+        )
+        return self.normalize_name(table_name) in tables
+
+    def _get_schema_names_query(self, connection, **kw):
+        if self._is_sql_fallback_requested(**kw):
+            return super()._get_schema_names_query(connection, **kw)
+        return "/*snapshot execution*/ " + super()._get_schema_names_query(connection, **kw)
+
+    @reflection.cache
+    def _get_columns(self, connection, table_name, schema=None, **kw):
+        if self._is_sql_fallback_requested(**kw):
+            return super()._get_columns(connection, table_name, schema, **kw)
+
+        tables = self._tables_for_schema(connection, schema=schema, table_name=table_name)
+        if len(tables) != 1:
+            return []
+
+        # get_columns_sql originally returned all columns of all tables if table_name is None,
+        # we follow this behavior here for compatibility. However, the documentation for Dialects
+        # does not mention this behavior:
+        # https://docs.sqlalchemy.org/en/13/core/internals.html#sqlalchemy.engine.interfaces.Dialect
+        quoted_schema_string = self.quote_string_value(tables[0].table_schem)
+        quoted_table_string = self.quote_string_value(tables[0].table_name)
+        sql_statement = "/*snapshot execution*/ {query}".format(query=self.get_column_sql_query_str())
+        sql_statement = sql_statement.format(schema=quoted_schema_string, table=quoted_table_string)
+        response = connection.execute(sql_statement)
+
+        return list(response)
+
+    @reflection.cache
+    def _get_pk_constraint(self, connection, table_name, schema=None, **kw):
+        if self._is_sql_fallback_requested(**kw):
+            return super()._get_pk_constraint(connection, table_name, schema, **kw)
+
+        conn = connection.engine.raw_connection()
+        schema = self._get_schema_for_input_or_current(connection, schema)
+        table_name = self.denormalize_name(table_name)
+        with conn.cursor().primaryKeys(table=table_name, schema=schema) as cursor:
+            pkeys = []
+            constraint_name = None
+            for row in cursor:
+                table, primary_key, constraint = row[2], row[3], row[5]
+                if table != table_name and table_name is not None:
+                    continue
+                pkeys.append(self.normalize_name(primary_key))
+                constraint_name = self.normalize_name(constraint)
+        return {'constrained_columns': pkeys, 'name': constraint_name}
+
+    @reflection.cache
+    def _get_foreign_keys(self, connection, table_name, schema=None, **kw):
+        if self._is_sql_fallback_requested(**kw):
+            return super()._get_foreign_keys(connection, table_name, schema, **kw)
+
+        # Need to use a workaround, because SQLForeignKeys functions doesn't work for an unknown reason
+        tables = self._tables_for_schema(
```

Comment:
Needs to be checked in another PR

Reply:
See https://github.com/exasol/sqlalchemy-exasol/issues/139